surface/CMakeLists.txt

```diff
@@ -45,9 +45,9 @@ if(build)
             src/vtk_smoothing/vtk_mesh_smoothing_windowed_sinc.cpp)
         
         if("${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}" VERSION_LESS "6.0")
-            set(VTK_SMOOTHING_TARGET_LINK_LIBRARIES vtkCommon vtkWidgets vtkGraphics)
+            set(VTK_SMOOTHING_TARGET_LINK_LIBRARIES pcl_io vtkCommon vtkWidgets vtkGraphics)
```

Comment:
Before adding this here, we need to report to `pcl_io` also as an optional dependency. The check now needs to be "bumped"  from simply finding VTK, to now ensuring `pcl_io` is built and also checking VTK is present.

Reply:
```diff
diff --git a/surface/CMakeLists.txt b/surface/CMakeLists.txt
index 719e88d41..1264fb631 100644
--- a/surface/CMakeLists.txt
+++ b/surface/CMakeLists.txt
@@ -1,10 +1,11 @@
 set(SUBSYS_NAME surface)
 set(SUBSYS_DESC "Point cloud surface library")
 set(SUBSYS_DEPS common search kdtree octree)
+set(SUBSYS_OPT_DEPS io qhull)
 
 set(build TRUE)
 PCL_SUBSYS_OPTION(build "${SUBSYS_NAME}" "${SUBSYS_DESC}" ON)
-PCL_SUBSYS_DEPEND(build "${SUBSYS_NAME}" DEPS ${SUBSYS_DEPS} OPT_DEPS qhull)
+PCL_SUBSYS_DEPEND(build "${SUBSYS_NAME}" DEPS ${SUBSYS_DEPS} OPT_DEPS ${SUBSYS_OPT_DEPS})
 
 PCL_ADD_DOC("${SUBSYS_NAME}")
 
@@ -24,8 +25,9 @@ if(build)
             src/convex_hull.cpp
             )
     endif(QHULL_FOUND)
-    
-    if (VTK_FOUND AND NOT ANDROID)
+
+    PCL_GET_SUBSYS_STATUS(IO_STATUS io)
+    if (IO_STATUS AND VTK_FOUND AND NOT ANDROID)
         set(VTK_USE_FILE "${VTK_USE_FILE}" CACHE INTERNAL "VTK_USE_FILE")
         include("${VTK_USE_FILE}")
         
   
```